src/Hedgehog.Experimental/Gen.fs

```diff
@@ -525,33 +548,35 @@ module GenX =
               return values.GetValue index |> unbox
             }
 
+        | Shape.Collection s ->
+            s.Accept {
+              new ICollectionVisitor<Gen<'a>> with
+              member _.Visit<'collection, 'element when 'collection :> System.Collections.Generic.ICollection<'element>> () =
+                match typeShape with
+                | Shape.Poco (:? ShapePoco<'a> as shape) ->
+                  gen {
+                    let! collection = genPoco shape
+                    let collection = collection |> unbox<System.Collections.Generic.ICollection<'element>>
+                    if canRecurse typeof<'element> then
+                      let recursionDepth = incrementRecursionDepth typeof<'element>
+                      let! seqRange = Gen.integral config.SeqRange
+                      for _ in [0 .. seqRange] do
+                        let! element = autoInner config recursionDepth
+                        collection.Add element
```

Comment:
You are mixing calls to `bind` with mutation.  This will reproduce the shrinking bug described in https://github.com/hedgehogqa/fsharp-hedgehog/issues/157.

Can you first create a test that fails because of this (like I did in PRs #51, #53, and #54)?

Then for the fix, I think you might have to create the elements before creating the `ICollection<_>`.

Reply:
D'oh!

Reply:
We are getting so much mileage out of `ListGen.sequence/traverse` :D